consts.php

```diff
@@ -10,6 +10,7 @@
 /** Repositories which must not have installer included in the matrix, even though their type suggests they should */
 const NO_INSTALLER_REPOS = [
     'silverstripe-config',
+    'doorman',
```

Comment:
Note the PHPDoc. "even though their type suggests they should".
This refers to the "type" in supported modules json, I believe - i.e. config is in the "supported modules" category but shouldn't have installer installed.

I don't think this will affect anything in "misc"

Reply:
It fixes the issue though, without this it will replicate the issue seen in CI

Reply:
doorman should get a matrix the same way other misc repos like [`silverstripe/silverstripe-standards`](https://github.com/silverstripe/silverstripe-standards/actions/runs/12227749460/job/34104918474) do. This is a workaround if it works, but it doesn't fix the bug.

Reply:
It even _did_ work that way as recently as last week: https://github.com/silverstripe/doorman/actions/runs/12120914500/job/33790686085

Reply:
I think we need this fix as is, because the 'fallback' when no installer branch can be found is used PHP versions for CMS 4 e.g. php 7.4-8.1

```php
    private function getListOfPhpVersionsByBranchName(): array
    {
        $branch = $this->getInstallerBranch();
        return MetaData::PHP_VERSIONS_FOR_CMS_RELEASES[$branch] ?? MetaData::PHP_VERSIONS_FOR_CMS_RELEASES['4'];
    }
```

I used NO_INSTALLER_REPOS because that allows us to presumably use the new silverstripe/doorman entry added to repositories.json to be used i.e. BranchLogic::getCmsMajor()

```php
    private function getInstallerBranch(): string
    {
        $version = str_replace('.x-dev', '', $this->installerVersion);
        if (in_array($this->repoName, NO_INSTALLER_REPOS) || (!in_array($this->repoName, FORCE_INSTALLER_REPOS) && isset($this->repoData['type']) && $this->repoData['type'] === 'recipe')) {
            $cmsMajor = BranchLogic::getCmsMajor($this->repoData, $this->branch, $this->getComposerJsonContent()) ?: MetaData::LOWEST_SUPPORTED_CMS_MAJOR;
            if (preg_match('#^[1-9]\.([0-9]+)$#', $this->branch, $matches)) {
                $version = sprintf('%d.%d', $cmsMajor, $matches[1]);
            } else {
                $version = $cmsMajor;
            }
        }

        return $version;
    }
```

silverstripe/silverstripe-standards can use the fallback because it does not mind what version of CMS is used, however doorman does as it has a minimum php version of 8.3 as it is specifically a CMS 6 branch

Even if we changed the fallback to CMS 5 and got rid of the change to NO_INSTALLER_REPOS, we'll still end up with a matrix that includes php 8.1-8.3 which are not all compatible

Reply:
> I think we need this fix as is, because the 'fallback' when no installer branch can be found is used PHP versions for CMS 4 e.g. php 7.4-8.1

Maybe the fallback should check the PHP composer constraint instead? That would solve for this and anything else similar in the future.

Alternatively maybe everything in `misc` should automatically be considered to be a `NO_INSTALLER_REPO`

> silverstripe/silverstripe-standards can use the fallback because it does not mind what version of CMS is used, however doorman does as it has a minimum php version of 8.3 as it is specifically a CMS 6 branch

We shouldn't be thinking of either of those in terms of a CMS version, since neither of them have dependencies on modules. I think what you meant was "`silverstripe/silverstripe-standards` can use the fallback because its minimum PHP version is within the range of PHP versions allowed by CMS 4 which is currently used as the fallback" and if that's the case, my suggestions above seem appropriate.

Reply:
Have gone for the fallback to PHP version in composer.json option